ruby/tests/basic_proto2.rb

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/ruby
+
+# basic_test_pb.rb is in the same directory as this test.
+$LOAD_PATH.unshift(File.expand_path(File.dirname(__FILE__)))
+
+require 'basic_test_proto2_pb'
+require 'common_tests'
+require 'google/protobuf'
+require 'json'
+require 'test/unit'
+
+# ------------- generated code --------------
+
+module BasicTestProto2
+  pool = Google::Protobuf::DescriptorPool.new
+  pool.build do
+    add_file "test_proto2.proto", syntax: :proto2 do
+      add_message "BadFieldNames" do
+        optional :dup, :int32, 1
+        optional :class, :int32, 2
+        optional :"a.b", :int32, 3
+      end
+    end
+  end
+
+  BadFieldNames = pool.lookup("BadFieldNames").msgclass
```

Comment:
I looked over everything once more, and it all looks good. I just had one question about this test: what is it intending to test? Some of those field names look problematic for Ruby but are valid field names for protobuf ("dup") while some of them are just invalid names period ("a.b"). Shouldn't this be throwing an error? What is this test trying to test.

Reply:
This proto setup is actually copied over from the proto3 test in `basic.rb` https://github.com/protocolbuffers/protobuf/blob/master/ruby/tests/basic.rb#L79. We had just copied over the test setup for proto3 from `basic.rb` and repurposed it for proto2.